client/sites/settings/v2/controller.tsx

```diff
@@ -0,0 +1,35 @@
+import { isEnabled } from '@automattic/calypso-config';
+import page from '@automattic/calypso-router';
+import {
+	SETTINGS_DATABASE,
+	SETTINGS_PERFORMANCE,
+	SETTINGS_SFTP_SSH,
+} from 'calypso/sites/components/site-preview-pane/constants';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import type { Context as PageJSContext } from '@automattic/calypso-router';
+
+export function redirectToHostingDashboardBackportIfEnabled( feature: string | undefined ) {
+	return ( context: PageJSContext, next: () => void ) => {
+		const state = context.store.getState();
+		const site = getSelectedSite( state );
+
+		if ( isEnabled( 'dashboard/v2/backport/site-settings' ) ) {
+			let route = '';
+			switch ( feature ) {
+				case SETTINGS_DATABASE:
+					route = '/database';
```

Comment:
This mapping from the old page to the new page, when an equivalent page exists, is nice. But I also think it'd be ok to just redirect everyone to the same settings overview page. Most new views don't have an equivalent in the old UI, so we have to be ok with some pages redirecting to the top-level. So I don't know if the extra code is worth it, and we could just redirect everyone to the settings overview.

I've tested though and it works well though.